src/App.js

```diff
@@ -1,28 +1,55 @@
 import './App.css'
-import Oggetti from './Components/OggettiComponents/Oggetti'
-import AllCategories from './Components/Filter/AllCategories'
-import Cart from './Components/Cart/Cart'
+import React, { useState, useEffect } from 'react'
+
 import Header from './Components/HeaderContent'
+import Shopping from './pages/Shopping'
+import Login from './pages/Login'
+import Register from './pages/Register'
+import Logout from './pages/Logout'
+
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
+import { checkShowMe } from './utils/api'
 
 function App() {
-  return (
-    <div>
-      <header>
-        <nav className='header_content'>
-          <Header />
-        </nav>
-      </header>
+  const [user, setUser] = useState(undefined);
 
-      <div className='aside'>
-        <AllCategories />
-        <Cart />
-      </div>
+  useEffect(() => {
+    async function getLoggedInUser() {
+      const user = await checkShowMe()
+      if (user) {
+        setUser(user)
+      }
+    }
+    getLoggedInUser()
+  }, [])
```

Comment:
Due to the empty array that we're passing it, this `useEffect` will only run the **first** time that the component loads: https://react.dev/reference/react/useEffect#useeffect

(In the video I notice that it's getting called twice, it turns out this is because of a special thing React does when running in development mode (aka running locally) see [the docs for more info](https://react.dev/reference/react/useEffect#my-effect-runs-twice-when-the-component-mounts), but will go away when you deploy to production)